plugins/gcov/gcov.rake

```diff
@@ -154,67 +154,266 @@ if PROJECT_USE_DEEP_DEPENDENCIES
+
+    if html_enabled
+      # Determine the HTML report file name.
+      artifacts_file_html = GCOV_ARTIFACTS_FILE_HTML
+      if !(opts[:gcov_html_artifact_filename].nil?)
+        artifacts_file_html = File.join(GCOV_ARTIFACTS_PATH, opts[:gcov_html_artifact_filename])
+      end
+
+      args += "--html-details " if !(opts[:gcov_html_report_type].nil?) && (opts[:gcov_html_report_type] == 'detailed')
+      args += "--html-title \"#{opts[:gcov_html_title]}\" " unless opts[:gcov_html_title].nil?
+      args += "--html-absolute-paths " if !(opts[:gcov_html_absolute_paths].nil?) && opts[:gcov_html_absolute_paths]
+      args += "--html-encoding \"#{opts[:gcov_html_encoding]}\" " unless opts[:gcov_html_encoding].nil?
+
+      [:gcov_html_medium_threshold, :gcov_html_high_threshold].each do |opt|
+        args += "--#{opt.to_s.gsub('_','-').sub(/:?gcov-/,'')} #{opts[opt]} " unless opts[opt].nil?
+      end
+
+      # The following option must be appended last for gcovr version <= 4.2 to properly work.
+      args += "--html #{use_output_option ? "--output " : ""} \"#{artifacts_file_html}\" "
+    end
+
     return args
   end
 
-  desc 'Create gcov code coverage html/xml report (must run ceedling gcov first).'
+
+  # Generate a text report.
+  def generate_text_report(opts, args_common)
+    args_text = ""
+    message_text = "Creating a gcov text report"
+
+    if !(opts[:gcov_text_artifact_filename].nil?)
+      artifacts_file_txt = File.join(GCOV_ARTIFACTS_PATH, opts[:gcov_text_artifact_filename])
+      args_text += "--output \"#{artifacts_file_txt}\" "
+      message_text = message_text + " in '#{GCOV_ARTIFACTS_PATH}'... "
+    else
+      message_text += "... "
+    end
+
+    print message_text
+    STDOUT.flush
+
+    # Generate the text report.
+    run_gcovr(args_common + args_text)
+  end
+
+
+  # Get the gcovr version number as components.
+  # Returns [major, minor].
+  def get_gcovr_version()
+    gcovr_version_number_major = 0
+    gcovr_version_number_minor = 0
+    gcovr_version_info = %x|gcovr --version|
+    version_number_match_data = gcovr_version_info.match(/gcovr ([0-9]+)\.([0-9]+)/)
+
+    if !(version_number_match_data.nil?) && !(version_number_match_data[1].nil?) && !(version_number_match_data[2].nil?)
+        gcovr_version_number_major = version_number_match_data[1].to_i
+        gcovr_version_number_minor = version_number_match_data[2].to_i
+    end
+
+    return gcovr_version_number_major, gcovr_version_number_minor
+  end
+
+
+  # Output the shell result to the console.
+  def print_shell_result(shell_result)
+    puts "Done in %.3f seconds." % shell_result[:time]
+
+    if !(shell_result.nil?) && !(shell_result[:output].nil?) && (shell_result[:output].length > 0)
+      puts shell_result[:output]
+    end
+  end
+
+
+  # Run gcovr with the given arguments.
+  def run_gcovr(args)
+    command = @ceedling[:tool_executor].build_command_line(TOOLS_GCOV_POST_REPORT, [], args)
```

Comment:
Here to run gcovr you are using executable from gcov_post_report defined in yml file.
To detect gcovr version you are using `gcovr_version_info = %x|gcovr --version|`
It means if I set:
```
:gcov_post_report:
  :executable: my_custom_gcovr
```
the version will be detected from incorrect gcovr?

Reply:
@CezaryGapinski Thank you for your continued thorough review. You are correct that I need to fix the way the gcovr version is detected. Let's not yet complete the pull request.

Reply:
@CezaryGapinski  I pushed changes to fix the problem you found with getting the version info. What other issues can you find?